ImputationPipeline/ScoringPart.wdl

```diff
@@ -230,6 +255,160 @@ task ScoreVcf {
 	}
 }
 
+task AddInteractionTermsToScore {
+	input {
+		File vcf
+		File interaction_weights
+		File scores
+		File? sites
+		String basename
+		SelfExclusiveSites? self_exclusive_sites # The interaction term will only be added in no more than selfExclusiveSites.maxAllowed of the
+																					 # effect alleles listed in SelfExclusizeSites.sites is observed
+
+		Float mem = 8
+		Int block_buffer=10000000
+	}
+
+	Int disk_space =  3*ceil(size(vcf, "GB")) + 20
+
+	command <<<
+
+		tabix ~{vcf}
+
+		python3 << "EOF"
+		from cyvcf2 import VCF
+		import pandas as pd
+		import csv
+
+		vcf = VCF("~{vcf}", lazy=True)
+		samples = vcf.samples
+
+		def add_allele_to_count(site, allele, dictionary):
+			if site in dictionary:
+				dictionary[site][allele]=[0]*len(samples)
+			else:
+				dictionary[site]={allele:[0]*len(samples)}
+
+		interactions_allele_counts = dict()
+		interactions_dict = dict()
+		positions = set()
+		if ~{if defined(sites) then "True" else "False"}:
+			with open("~{sites}") as f_sites:
+				sites = {s.strip() for s in f_sites}
+		else:
+			sites = {}
+		with open("~{interaction_weights}") as f:
+			for line in csv.DictReader(f, delimiter='\t'):
+				site_1 = line['id_1']
+				site_2 = line['id_2']
+				if len(sites) == 0 or site_1 in sites and site_2 in sites:
```

Comment:
I always like to set parentheses when having an "a or b and c" condition for clarity, because different languages sometimes process them differently (I think), but feel free to leave it for this merge, we can always work on aesthetics later